cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -662,6 +664,131 @@ var _ = Describe("Operation Controller", func() {
 				deleteTestResources(ctx, dbQueries, resourcesToBeDeleted)
 
 			})
+			It("Checks whether the user updated tls-certificate verification maps correctly from database to cluster secret", func() {
+				By("Close database connection")
+				defer dbQueries.CloseDatabase()
+				defer testTeardown()
+
+				_, dummyApplicationSpecString, err := createDummyApplicationData()
+				Expect(err).To(BeNil())
+
+				gitopsEngineCluster, _, err := dbutil.GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID(ctx, string(kubesystemNamespace.UID), dbQueries, logger)
+				Expect(gitopsEngineCluster).ToNot(BeNil())
+				Expect(err).To(BeNil())
+
+				By("creating a cluster credential with tls-cert-verification set to true")
+
+				clusterCredentials := &db.ClusterCredentials{
+					Clustercredentials_cred_id:  "test-clustercredentials_cred_id-1",
+					Host:                        "test-host",
+					Kube_config:                 "test-kube_config",
+					Kube_config_context:         "test-kube_config_context",
+					Serviceaccount_bearer_token: "test-serviceaccount_bearer_token",
+					Serviceaccount_ns:           "test-serviceaccount_ns",
+					AllowInsecureSkipTLSVerify:  bool(managedgitopsv1alpha1.TLSVerifyStatusTrue),
```

Comment:
As with above, it would be good to also test the `TLSVerifyStatusFalse` case with this test. You can use `DescribeTable` to run the test with both values (for example), so there isn't any code duplication.